agent/config/builder.go

```diff
@@ -715,6 +715,8 @@ func (b *Builder) Build() (rt RuntimeConfig, err error) {
 		DNSSOA:                soa,
 		DNSUDPAnswerLimit:     b.intVal(c.DNS.UDPAnswerLimit),
 		DNSNodeMetaTXT:        b.boolValWithDefault(c.DNS.NodeMetaTXT, true),
+		DNSUseCache:           b.boolVal(c.DNS.UseCache),
+		DNSCacheMaxAge:        b.durationVal("dns_config.cache_max_age", c.DNS.CacheMaxAge),
```

Comment:
I don't have a strong opinion on this. To me it is important that the HTTP and DNS APIs behave the same way to avoid confusion.
`max_stale` currently doesn't have a 'if-error' semantic. It fells like piggybacking to use it for this purpose. 
I feel that `cache_max_age` and `max_stale` are already redundant. The end user does not care about what cache induced the staleness but rather wants results 'at most this stale' and 'at most this stale if an error happens'.